File: 34.py
def factorial(f):
    if f == 0:
        return 1
    elif f == 1:
        return 1
    else:
        return f*factorial(f-1)

factorials = []
for i in range(0,10):
    factorials.append(factorial(i))

totalSum = 0
max = 7*factorials[9]
i = 10
while i < max:
    sum = 0
    for c in str(i):
        n = ord(c) - ord("0")
        f = factorials[n]
        if f > i:
            break
        sum += f
    if sum == i:
        print "\tmatch: " + str(i)
        totalSum += sum
    i += 1
print "totalsum: " + str(totalSum)

File: 34.c
#include <stdio.h>

int factorial(int f)
{
	if (f == 0) {
		return 1;
	}

	return (f == 1) ? 1 : (factorial(f-1) * f);
}

int main(void)
{
	int factorials[10];
	unsigned int i;
	for (i = 0; i < 10; i++) {
		factorials[i] = factorial(i);
	}

	unsigned int totalSum = 0, sum, j;
	for (i = 10; i < 8*factorials[9]; i++) {
		sum = 0;
		j = i;
		while (j > 0) {
			sum += factorials[j%10];
			j /= 10;
		}
		if (sum == i) {
			printf("%d\n", i);
			totalSum += i;
		}
	}
	printf("total sum: %d\n", totalSum);

	return 0;
}

